### ISSUE TITLE
Native ESM

### ISSUE THREAD

**Issue by PhearZero** [algorand/js-algorand-sdk#819]
##  ❌ Problem

The following errors due to `node.js` resolving the `commonjs` bundle while using native `esm` modules. 

Reproduction: https://github.com/PhearZero/js-algorand-sdk-node-18-module/tree/main

```javascript
import { Indexer } from 'algosdk'
```

### 🤔Reasoning:

While modern bundlers have robust resolvers, it would be good to have first class support for `esm` without third party tools. This will also improve third party bundlers ability to tree shake. 

## ✅ Solution

Convert package to `esm`.

- Package type "commonjs" loads the `./dist/cjs/index.js` entrypoint. ([fixed example using fork](https://github.com/PhearZero/js-algorand-sdk-node-18-module/tree/fix/cjs))
- Package type "module" loads the `./dist/esm/index.js` entrypoint. ([fixed example using fork](https://github.com/PhearZero/js-algorand-sdk-node-18-module/tree/fix/esm))

## 📦Dependencies

This requires moving to `tsx` and slight refactoring of the codebase. 

##  ‼️ Urgency

Very low priority

**Comment by jasonpaulos** [comment]
Hi @PhearZero, I am under the impression that having this line in our `package.json` is enough to provide ESM support: https://github.com/algorand/js-algorand-sdk/blob/7469c41ab4a54dec44a9ee55da57d54bafd444be/package.json#L6

Can you please share more details about your environment (e.g. operating system, node version) and the exact error you're receiving?

**Comment by PhearZero** [comment]
> Hi @PhearZero, I am under the impression that having this line in our package.json is enough to provide ESM support:

This is true for most bundlers but it's actually not standardized, just a widely adopted key. The actual `package.json` standard does not include `module` but `exports`.  The previous ESM bundles where an attempt at "future proofing" but not compliant with the current specification.  

- [Node.js Exports Documentation](https://nodejs.org/api/packages.html#package-entry-points)
- [Typescript ESM](https://www.typescriptlang.org/docs/handbook/esm-node.html)
- [Browser Support](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules)

> Can you please share more details about your environment (e.g. operating system, node version) and the exact error you're receiving?

Should be any node/os combination that supports `esm`.  **Node >= v14** 

Here is the reproduction repo using Node v18: https://github.com/PhearZero/js-algorand-sdk-node-18-module

## Steps:

```bash
npm init
npm install algosdk
```

Set package.json `type` to `module`
```json
{
 "type": "module"
}
```

Import module

```javascript
import {Indexer} from 'algosdk'

// Error does not provide default export

```

**Comment by jasonpaulos** [comment]
Addressed in #836
